src/FraudSession.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Drupal\commerce_bluesnap;
+
+use Drupal\Core\TempStore\PrivateTempStoreFactory;
+
+/**
+ * Client for making requests to the Card/Wallet Transactions API.
+ */
+class FraudSession {
+
+  /**
+   * The shared temporary storage for commerce_bluesnap.
+   *
+   * @var \Drupal\Core\TempStore\SharedTempStore
+   */
+  protected $tempStore;
+
+  /**
+   * Constructs a FraudSession object.
+   *
+   * @param \Drupal\Core\TempStore\PrivateTempStoreFactory $temp_store_factory
+   *   The tempstore factory.
+   */
+  public function __construct(PrivateTempStoreFactory $temp_store_factory) {
+    $this->tempStore = $temp_store_factory->get('commerce_bluesnap');
+  }
+
+  /**
+   * Returns fraud session id
+   *
+   * @return string
+   *   Bluesnap fraud session ID
+   */
+  public function get() {
+    if (!$this->tempStore->get('fraud_session_id')) {
+      $this->tempStore->set('fraud_session_id', $this->generate());
+    }
+    return $this->tempStore->get('fraud_session_id');
+  }
```

Comment:
**CHANGE REQUEST**

*Performance*

You can avoid unnecessary function calls. You're calling the `get` function twice, store it in a variable to avoid that.

*Readability*

It's also good to separate pieces of code with line breaks.

```
public function get() {
  $session_id = $this->tempStore->get('fraud_session_id');
  if (!$session_id) {
    $this->tempStore->set('fraud_session_id', $this->generate());
  }

  return $session_id;
}
```

Reply:
@krystalcode I see a problem here. We are setting session ID if its empty right? and we are not overriding the session_id variable. So there is a chance that this method returning empty value if session ID was not set at all? Do I make sense?

Reply:
@safalliajoseph My point was to avoid unnecessary function calls. Please adjust the code as needed if I have a mistake, all the code I post in comments is untested off-the-top-of-my-head.

```
public function get() {
  $session_id = $this->tempStore->get('fraud_session_id');
  if (!$session_id) {
    $session_id = $this->generate();
    $this->tempStore->set('fraud_session_id', $session_id);
  }

  return $session_id;
}
```

Reply:
Corrected